intro-HPC/ch_running_batch_jobs.tex

```diff
@@ -40,6 +40,331 @@ \chapter{Running batch jobs}
+
+\subsection{Environment Variables}
+
+The program environment on the \hpc is controlled by pre-defined settings,
+which are stored in environment (or shell) variables.
+
+You can use shell variables to store data, set configuration options and
+customise the environment on the \hpc. The default shell
+on the \hpc is Bash (Bourne Again Shell) and can be used for the
+following purposes:
+
+\begin{enumerate}
+\item  Configure look and feel of the shell.
+\item  Setup terminal settings depending on which terminal you're using.
+\item  Set the search path for running executables.
+\item  Set environment variables as needed by programs.
+\item  Set convenient abbreviations for heavily used values.
+\item  Run commands that you want to run whenever you log in or log out.
+\item  Setup aliases and/or shell function to automate tasks to save typing and time.
+\item  Changing the bash prompt.
+\item  Setting shell options.
+\end{enumerate}
+
+The environment variables are typically set at login by a script, whenever you
+connect to the \hpc. These pre-defined variables usually impact the run time
+behaviour of the programs that we want to run.
+
+All the software packages that are installed on the \hpc cluster require
+different settings. These packages include compilers, interpreters,
+mathematical software such as MATLAB and SAS, as well as other applications and
+libraries.
+
+In order to administer the active software and their environment variables, a
+\strong{\emph{``module''}} package has been developed, which:
+
+\begin{enumerate}
+\item  Activates or deactivates \underbar{software packages} and their dependencies.
+\item  Allows setting and unsetting of \underbar{environment variables}, including adding and deleting entries from database-type environment variables.
+\item  Does this in a \underbar{shell-independent} fashion (necessary information is stored in the accompanying module configuration file).
+\item  Takes care of \underbar{versioning aspects}: For many libraries, multiple versions are installed and maintained. The module system also takes care of the versioning of software packages in case multiple versions are installed. For instance, it does not allow multiple versions to be loaded at same time.
+\item  Takes care of \underbar{dependencies}: Another issue arises when one considers library versions and the dependencies they create. Some software requires an older version of a particular library to run correctly (or at all). Hence a variety of version numbers is available for important libraries.
+\end{enumerate}
+
+This is all managed with the ``\emph{module}'' command, which is explained in the next sections.
+
+\subsection{Available modules}
+
+A large number of software packages are installed on the \hpc clusters. A
+list of all currently available software can be obtained by typing:
+
+\begin{prompt}
+%\shellcmd{module av}%
+\end{prompt}
+
+or by typing it in full:
+
+\begin{prompt}
+%\shellcmd{module available}%
+\end{prompt}
```

Comment:
There are multiple equivalent commands: `module available` (that's the long version), `module avail`, `module av` also work.

With Lmod (UGent-specific), also `ml av` & the longer ones using `ml` rather than `module` work.

Please try to clarify that, but keep it short...